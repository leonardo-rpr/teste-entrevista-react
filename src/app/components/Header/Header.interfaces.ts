export interface HeaderInterface {
  onClick: any;
}
