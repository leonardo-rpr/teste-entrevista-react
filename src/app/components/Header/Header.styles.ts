import styled from 'styled-components';

export const Container = styled.header<{}>`
  display: flex;
  align-items: center;
  justify-content: flex-end;
  height: 50px;
  background: #f2295b;
`;
