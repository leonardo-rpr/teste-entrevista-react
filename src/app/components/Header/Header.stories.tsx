import React from 'react';
import { storiesOf } from '@storybook/react';
import Header from './Header';

storiesOf('Components', module).add('Header', () => <Header onClick={() => {}} />);
