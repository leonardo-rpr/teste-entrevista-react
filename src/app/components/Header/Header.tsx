import React from 'react';
import { FaSignOutAlt } from 'react-icons/fa';

import { Container, Button } from './Header.styles';
import { HeaderInterface } from './Header.interfaces';

const Header: React.FC<HeaderInterface> = ({ onClick }) => (
  <Container>
    <Button onClick={onClick}>
      <FaSignOutAlt size="20px" color="#FFF" />
    </Button>
  </Container>
);

export default Header;
