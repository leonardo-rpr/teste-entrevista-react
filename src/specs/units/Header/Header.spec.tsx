import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Header from '../../../app/components/Header';

afterEach(cleanup);

describe('Header', () => {
  it('Renders at the default state correctly', () => {
    const wrapper = render(<Header onClick={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  });
});
