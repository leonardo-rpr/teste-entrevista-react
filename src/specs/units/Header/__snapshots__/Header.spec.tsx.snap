// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders at the default state correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-bdVaJa iTJPqY"
      >
        <button
          class="sc-bwzfXH iTMuyI"
          type="button"
        >
          <svg
            color="#FFF"
            fill="currentColor"
            height="20px"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 512 512"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
            />
          </svg>
        </button>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-bdVaJa iTJPqY"
    >
      <button
        class="sc-bwzfXH iTMuyI"
        type="button"
      >
        <svg
          color="#FFF"
          fill="currentColor"
          height="20px"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(255, 255, 255);"
          viewBox="0 0 512 512"
          width="20px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
          />
        </svg>
      </button>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
